app/models/concerns/shiny_show_hide.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+# Common behaviour for URL slugs
+module ShinyShowHide
+  extend ActiveSupport::Concern
+
+  included do
+    # Validations
+
+    # rubocop:disable Style/RedundantSelf
+    validates :show_on_site,  inclusion: { in: [ true, false ] }
+    validates :show_in_menus, inclusion: { in: [ true, false ] } if self.respond_to?( :show_in_menus )
```

Comment:
I'm not clear if this is doing what's intended. What's the purpose of the `if self.respond_to?(:show_in_menus)` check?

Reply:
Not everything that uses the ShowHide concern has a show_in_menus attribute.

Reply:
My intention is to refactor this more extensively at some point; it seems like an obvious candidate for a polymorphic trait that can store the show/hide status for any instance of any object in a number of named scenarios (so I get something like section.pages.show_on( :menus).all at the far end). That's a way down the to-do list while I'm trying to focus on forms though, so this is a bit of interim tidying that took much less time to think about :slightly_smiling_face:

Reply:
> Not everything that uses the ShowHide concern has a show_in_menus attribute.

Because of the scope of this line 12, this may not work as intended.

The scope here means `self` will be the class only (e.g. `Blog`), and not an instance of the class (e.g. `Blog.new`).

If line 12 isn't working as intended, this may work instead (needs testing):

```diff
-validates :show_in_menus, inclusion: { in: [ true, false ] } if self.respond_to?( :show_in_menus )
+validates :show_in_menus, inclusion: { in: [ true, false ] } if self.column_names.include?( 'show_in_menus' )
```